src/Workspaces/Core/Portable/Rename/Renamer.cs

```diff
@@ -30,7 +30,11 @@ public static class Renamer
 
             optionSet ??= solution.Options;
 
-            return RenameSymbolAsync(solution, symbol, newName, optionSet, nonConflictSymbols: null, cancellationToken);
+            var resolution = await RenameSymbolAsync(solution, symbol, newName, optionSet, nonConflictSymbols: null, cancellationToken).ConfigureAwait(false);
+            if (resolution.ErrorMessage != null)
+                throw new ArgumentException(resolution.ErrorMessage);
```

Comment:
Why do we use `ArgumentException` here to surface the error message? Could use comment.

Reply:
sure will add.

Reply:
done.